dao-contracts/src/repo_voter.rs

```diff
@@ -0,0 +1,118 @@
+use casper_dao_modules::{vote::Vote, voting::Voting, GovernanceVoting, VotingId};
+use casper_dao_utils::{
+    casper_dao_macros::{casper_contract_interface, Instance},
+    casper_env::get_block_time,
+    consts, Address,
+};
+use casper_types::{bytesrepr::Bytes, runtime_args, RuntimeArgs, U256};
+
+use crate::VariableRepositoryContractCaller;
+
+#[casper_contract_interface]
+pub trait RepoVoterContractInterface {
+    fn init(&mut self, variable_repo: Address, reputation_token: Address);
+    fn create_voting(
+        &mut self,
+        variable_repo_to_edit: Address,
+        key: String,
+        value: Bytes,
+        activation_time: Option<u64>,
+        stake: U256,
+    );
+    fn vote(&mut self, voting_id: VotingId, choice: bool, stake: U256);
+    fn finish_voting(&mut self, voting_id: VotingId);
+    fn get_dust_amount(&self) -> U256;
+    fn get_variable_repo_address(&self) -> Address;
+    fn get_reputation_token_address(&self) -> Address;
+    fn get_voting(&self, voting_id: U256) -> Voting;
+    fn get_vote(&self, voting_id: U256, address: Address) -> Vote;
+    fn get_voters(&self, voting_id: U256) -> Vec<Option<Address>>;
+}
+
+#[derive(Instance)]
+pub struct RepoVoterContract {
+    voting: GovernanceVoting,
+}
+
+impl RepoVoterContractInterface for RepoVoterContract {
+    fn init(&mut self, variable_repo: Address, reputation_token: Address) {
+        self.voting.init(variable_repo, reputation_token);
+    }
+
+    fn create_voting(
+        &mut self,
+        variable_repo_to_edit: Address,
+        key: String,
+        value: Bytes,
+        activation_time: Option<u64>,
+        stake: U256,
+    ) {
+        let variable_repo_package_hash = *self
+            .voting
+            .get_variable_repo_address()
+            .as_contract_package_hash()
+            .unwrap();
+        let repo_caller = VariableRepositoryContractCaller::at(variable_repo_package_hash);
+        let informal_voting_time = repo_caller.get_variable(consts::INFORMAL_VOTING_TIME);
+        let informal_voting_quorum = repo_caller.get_variable(consts::INFORMAL_VOTING_QUORUM);
+        let formal_voting_time = repo_caller.get_variable(consts::FORMAL_VOTING_TIME);
+        let formal_voting_quorum = repo_caller.get_variable(consts::FORMAL_VOTING_QUORUM);
+        let minimum_governance_reputation =
+            repo_caller.get_variable(consts::MINIMUM_GOVERNANCE_REPUTATION);
+
+        let voting = Voting {
+            voting_id: self.voting.votings_count.get(),
```

Comment:
informal_voting_id is needed for formal voting to point into previous one
also voting_id = informal_voting_id means that this voting is informal - we could add an enum to voting, but it would be redundant